iib/web/models.py

```diff
@@ -1097,17 +1095,37 @@ def get_mutable_keys(self):
         return rv
 
 
-class RequestMergeIndexImage(Request, RequestIndexImageMixin):
+class RequestMergeIndexImage(Request):
     """A "merge-index-image" build request."""
 
     __tablename__ = 'request_merge_index_image'
 
     id = db.Column(db.Integer, db.ForeignKey('request.id'), autoincrement=False, primary_key=True)
-    source_from_index_id = db.Column(db.Integer, db.ForeignKey('image.id'), nullable=True)
+    binary_image_id = db.Column(db.Integer, db.ForeignKey('image.id'), nullable=False)
+    binary_image_resolved_id = db.Column(db.Integer, db.ForeignKey('image.id'))
+    binary_image = db.relationship('Image', foreign_keys=[binary_image_id], uselist=False)
+    binary_image_resolved = db.relationship(
+        'Image', foreign_keys=[binary_image_resolved_id], uselist=False
+    )
+
+    deprecation_list = db.relationship('Image', secondary=BundleDeprecation.__table__)
+
+    index_image_id = db.Column(db.Integer, db.ForeignKey('image.id'))
```

Comment:
What's this for?

Reply:
This is the pullspec of the index image generated by IIB. If it's overwritten, then it'll be the original pullspec, if it's not, then it'll be the one in the container registry repo IIB is configured to use.